src/new-components/base-button/base-button.tsx

```diff
@@ -111,6 +117,8 @@ export const BaseButton = polymorphicComponent<'div', BaseButtonProps>(function
                 styles[`variant-${variant}`],
                 styles[`tone-${tone}`],
                 styles[`size-${size}`],
+                width !== 'auto' && icon == null && align != null ? styles[`align-${align}`] : null,
```

Comment:
Since `align` is only meant to be used with the `width="full"` prop, it might be helpful to guard against them with something like this (haven't tried if it works though):

```TS
type AlignmentProps = {
	width: 'full'
	align?: 'left' | 'center' | 'right'
}

type AutoWidthProps = {
	width?: 'auto'
	align?: never
}

type LabelledButtonProps = { ... } & (AlignmentProps | AutoWidthProps)
```

Reply:
I thought about it, and as you know, we already do something along those lines to take care of the icon-only vs. the full-visual-text-label cases, as these have even more striking differences. I was hesitant, though, to introduce this level of complexity in the types. But maybe is something that the library can contain.

Another alternative that I thought was to not introduce the `width` prop at all. And make the button be full width whenever the `align` property is provided. But this on the other hand seemed a bit obscure.

Thoughts?

Reply:
Yeah I don't think it should be omitted. It's not obvious `<Button align="center">` should be expanded just by looking at it 🤔 I think the explicitness of having `width="full"` in there would be more preferable.

Reply:
That's what I thought too.

I'll tinker the types a bit to see if I can achieve this without complicating things too much.

Reply:
Your suggestion seems to be working well. I'll validate a few things, but it looks like we'll make it.